stan/math/prim/scal/fun/value_of_rec.hpp

```diff
@@ -20,7 +20,7 @@ namespace math {
  * @param x Scalar to convert to double.
  * @return Value of scalar cast to a double.
  */
-template <typename T>
+template <typename T, typename = require_stan_scalar_t<T>>
```

Comment:
Just to double check, `require_stan_scalar_t` works for var, fvar, double, int, etc. Is that what you want here?

Reply:
This works fine as we have overloads for double, var and fvar. I only need this overload for int and anything else that matches that "etc". Since I am not sure what that could be I just went with the more general requirement. Do we have any more specific `require_...` I should use here?

Reply:
Nope I think this is fine